### PR TITLE
WMO: Added lighting and fog

### PR DESCRIPTION
**Pull request by fallenoak**
Slowly getting closer to matching the client's lighting.

### CODE REVIEW

File: src/lib/pipeline/wmo/group/index.js [hunk 1]
```diff
@@ -27,22 +32,58 @@ class WMOGroup extends THREE.Mesh {
...
+        colors[index * 3 + 1] = color.g / 255.0;
+        colors[index * 3 + 2] = color.b / 255.0;
+        alphas[index] = color.a / 255.0;
+      });
+    } else if (this.indoor) {
+      // Default indoor vertex color: rgba(0.5, 0.5, 0.5, 1.0)
+      data.MOVT.vertices.forEach(function(vertex, index) {
+        colors[index * 3] = 127.0/255.0;
+        colors[index * 3 + 1] = 127.0/255.0;
+        colors[index * 3 + 2] = 127.0/255.0;
```
Comment (timkurvers) [inline src/lib/pipeline/wmo/group/index.js]: Could you add some spacing here between the `/`'s operands?

File: src/lib/pipeline/wmo/group/index.js [hunk 2]
```diff
@@ -27,22 +32,58 @@ class WMOGroup extends THREE.Mesh {
...
     geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
+    geometry.addAttribute('normal', new THREE.BufferAttribute(normals, 3));
     geometry.addAttribute('uv', new THREE.BufferAttribute(uvs, 2));
 
+    // TODO: Perhaps it is possible to directly use a vec4 here? Currently, color + alpha is
+    // combined into a vec4 in the material's vertex shader. For some reason, attempting to
+    // directly use a BufferAttribute with a length of 4 resulted in incorrect ordering for the
+    // values in the shader.
+    geometry.addAttribute('color', new THREE.BufferAttribute(colors, 3));
+    geometry.addAttribute('alpha', new THREE.BufferAttribute(alphas, 1));
```
Comment (timkurvers) [inline src/lib/pipeline/wmo/group/index.js]: We probably want to tackle this at a later point? After this PR?

File: src/lib/pipeline/wmo/group/index.js (hunk 1 above)
Comment (fallenoak) [inline src/lib/pipeline/wmo/group/index.js]: Will do!

File: src/lib/pipeline/wmo/group/index.js (hunk 2 above)
Comment (fallenoak) [inline src/lib/pipeline/wmo/group/index.js]: Yeah, sounds good. I couldn't tell if the issue was THREE trying to munge attributes named 'color' in a weird way, or if perhaps the values were perhaps going like this: [0, 1, 2, 3] => [3, 0, 1, 2]

**Comment by timkurvers** [comment]
Fantastic stuff! :tada:

**Pull request merged**